pkg/cli/cli.go

```diff
@@ -1439,13 +1436,40 @@ func runConnectionWebSocketAgentCommand(flags *ConnectFlags, c *command, args []
 
 // Wrapper for implementing io.ReadWriteCloser of websocket.Conn
 type wsIoWrapper struct {
-	wsConn *websocket.Conn
-	pos    int
-	buf    []byte
+	wsConn   *websocket.Conn
+	pos      int
+	buf      []byte
+	stopPing chan interface{}
 }
 
 var _ io.ReadWriteCloser = (*wsIoWrapper)(nil)
 
+func newWsIoWrapper(c *command, wsConn *websocket.Conn) *wsIoWrapper {
+	stopPing := make(chan interface{})
+	go func(stopPing chan interface{}) {
```

Comment:
let's move this logic out of the `newWsIoWrapper` constructor. 

Suggest having this logic around a new struct: `wsPingSender`

```
type wsPingSender struct {
  wsConn *websocket.Conn
  delay time.Duration
  stopCh chan interface{}
}

func (w *wsPingSender) Start() error {
 // Add logic here
}
```

Reply:
Declared new type `wsPingSender` and changed `wsIoWrapper` to use it.

Reply:
Nice! thanks.